master/internal/api_experiment.go

```diff
@@ -51,6 +53,65 @@ import (
 
 var experimentsAddr = actor.Addr("experiments")
 
+// Catches information on active running experiments.
+type experimentAllocation struct {
+	Job      model.JobID
+	Pulling  bool
+	Running  bool
+	Starting bool
+}
+
+func (a *apiServer) enrichExperimentState(experiments []*experimentv1.Experiment) (
+	[]*experimentv1.Experiment, error,
+) {
+	// filter allocations by JobIDs on this page of experiments
+	jobFilter := make([]string, 0, len(experiments))
+	for _, exp := range experiments {
+		jobFilter = append(jobFilter, exp.JobId)
+	}
+
+	// get active experiments by JobID
+	tasks := []experimentAllocation{}
+	err := a.m.db.Query(
+		"get_active_allocations_by_job",
```

Comment:
i would rename these queries to `aggregate_allocation_state_by_job` or something to better express its cost.